File: backend/admin-system/src/main/java/top/flobby/admin/system/domain/entity/Region.java
package top.flobby.admin.system.domain.entity;

import java.time.LocalDateTime;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

/**
 * 区域表实体
 *
 * @author Code Generator
 * @date 2026-02-01
 */
@Getter
@Setter
@Entity
@Table(name = "t_region")
@EntityListeners(AuditingEntityListener.class)
public class Region {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    /**
     * 区域名称
     */
    @Column(name = "name", nullable = false, length = 100)
    private String name;

    /**
     * 区域编码
     */
    @Column(name = "region_code", nullable = false, length = 50)
    private String regionCode;

    /**
     * 父级编码
     */
    @Column(name = "parent_code", length = 50)
    private String parentCode = "0";

    /**
     * 层级：1-省，2-市，3-区县
     */
    @Column(name = "level", nullable = false)
    private Integer level = 1;

    /**
     * 排序
     */
    @Column(name = "sort_order")
    private Integer sortOrder = 0;

    /**
     * 状态：0-禁用，1-启用
     */
    @Column(name = "status", nullable = false)
    private Integer status = 1;

    /**
     * 创建时间
     */
    @CreatedDate
    @Column(name = "create_time", updatable = false)
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    @LastModifiedDate
    @Column(name = "update_time")
    private LocalDateTime updateTime;

    /**
     * 创建人
     */
    @Column(name = "create_by", length = 50)
    private String createBy;

    /**
     * 更新人
     */
    @Column(name = "update_by", length = 50)
    private String updateBy;

    /**
     * 删除标记：0-未删除，1-已删除
     */
    @Column(name = "deleted", nullable = false)
    private Integer deleted = 0;

}

File: frontend/src/api/region.ts
import request from '@/utils/request'

/**
 * 区域表API
 * @author Code Generator
 * @date 2026-02-01
 */

export interface RegionVO {
  /** 主键ID */
  id: number
  /** 区域名称 */
  name: string
  /** 区域编码 */
  regionCode: string
  /** 父级编码 */
  parentCode: string
  /** 层级：1-省，2-市，3-区县 */
  level: number
  /** 排序 */
  sortOrder: number
  /** 状态：0-禁用，1-启用 */
  status: number
  /** 创建时间 */
  createTime: string
  /** 更新时间 */
  updateTime: string
  /** 创建人 */
  createBy: string
}

export interface RegionDTO {
  /** 区域名称 */
  name: string
  /** 区域编码 */
  regionCode: string
  /** 父级编码 */
  parentCode?: string
  /** 层级：1-省，2-市，3-区县 */
  level: number
  /** 排序 */
  sortOrder?: number
  /** 状态：0-禁用，1-启用 */
  status: number
}

export interface RegionQuery {
  /** 区域名称 */
  name?: string
  /** 区域编码 */
  regionCode?: string
  /** 父级编码 */
  parentCode?: string
  /** 状态：0-禁用，1-启用 */
  status?: number
  /** 页码 */
  pageNum?: number
  /** 每页数量 */
  pageSize?: number
}

/**
 * 获取区域表列表
 */
export function listRegions() {
  return request<RegionVO[]>({
    url: '/system/regions',
    method: 'get'
  })
}

/**
 * 获取区域表详情
 */
export function getRegion(id: number) {
  return request<RegionVO>({
    url: `/system/regions/${id}`,
    method: 'get'
  })
}

/**
 * 创建区域表
 */
export function createRegion(data: RegionDTO) {
  return request<number>({
    url: '/system/regions',
    method: 'post',
    data
  })
}

/**
 * 更新区域表
 */
export function updateRegion(id: number, data: RegionDTO) {
  return request<void>({
    url: `/system/regions/${id}`,
    method: 'put',
    data
  })
}

/**
 * 删除区域表
 */
export function deleteRegion(id: number) {
  return request<void>({
    url: `/system/regions/${id}`,
    method: 'delete'
  })
}
